app/Services/Model/AttendeeService.php

```diff
@@ -469,4 +477,20 @@ public function doVirtualCheckin(Summit $summit, int $attendee_id): ?SummitAtten
             return $attendee;
         });
     }
+
+    public function doCheckIn(Summit $summit, String $qr_code): void
+    {
+        $this->tx_service->transaction(function() use($summit, $qr_code){
+
+            $fields        = SummitAttendeeBadge::parseQRCode($qr_code);
+            $ticket_number = $fields['ticket_number'];
```

Comment:
could we add validations here for the QR prefix ?
https://github.com/OpenStackweb/summit-api/blob/172a04a7407e7e96f7c08653429513ea606e0a7f/app/Services/Model/Imp/SponsorUserInfoGrantService.php#L120

Reply:
done